lib/publiccloud/openstack_client.pm

```diff
@@ -0,0 +1,78 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for OpenStack connection and authentication
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+package publiccloud::openstack_client;
+use Mojo::Base -base;
+use testapi;
+use utils qw(file_content_replace);
+
+use constant OS_CONFIG_FILE => '/root/.config/openstack/clouds.yaml';
+use constant CLOUD_INIT_FILE => '/root/mn_jeos.cloud-init';
+
+has auth_uri => sub { get_required_var('PUBLIC_CLOUD_OPENSTACK_AUTH_URI') };
+has project_name => sub { get_required_var('PUBLIC_CLOUD_OPENSTACK_PROJECT_NAME') };
+has project_id => sub { get_required_var('PUBLIC_CLOUD_OPENSTACK_PROJECT_ID') };
+has region => sub { get_var('PUBLIC_CLOUD_REGION', 'CustomRegion') };
```

Comment:
does it really make sense ? can we drop this param from here and terraform file ?

Reply:
> does it really make sense ? can we drop this param from here and terraform file ?

Good question, tried it manually. 
TF will just yield ` region              = (known after apply)`

Reply:
region is actually a mandatory parameter in the [terraform_aply command](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/provider.pm#L382). It's not needed, but if I don't put it here it will complain when doing terraform apply.